File: pixels-core/src/main/java/cn/edu/ruc/iir/pixels/core/PixelsVersion.java
package cn.edu.ruc.iir.pixels.core;

/**
 * pixels
 *
 * @author guodong
 */
public enum PixelsVersion
{
    V1(1);

    private int version;

    PixelsVersion(int version)
    {
        this.version = version;
    }

    public int getVersion()
    {
        return this.version;
    }

    public static PixelsVersion from(int version)
    {
        if (version == 1)
        {
            return PixelsVersion.valueOf("V1");
        }
        throw new IllegalArgumentException("Wrong version.");
    }

    public static boolean matchVersion(int otherVersion)
    {
        return otherVersion == V1.version;
    }

    public static PixelsVersion currentVersion()
    {
        return V1;
    }
}

File: cpp/pixels-reader/pixels-core/lib/PixelsVersion.cpp
//
// Created by liyu on 3/13/23.
//

#include "PixelsVersion.h"
#include "exception/InvalidArgumentException.h"

int PixelsVersion::getVersion() {
    return version;
}

PixelsVersion::PixelsVersion(int v) {
    version = v;
}

PixelsVersion::Version PixelsVersion::from(int v) {
    if (v == 1) {
        return V1;
    } else {
        throw InvalidArgumentException("Wrong pixels version. ");
    }
}

bool PixelsVersion::matchVersion(PixelsVersion::Version otherVersion) {
    return otherVersion == V1;
}

PixelsVersion::Version PixelsVersion::currentVersion() {
    return V1;
}




